operations/fetch.go

```diff
@@ -263,16 +270,26 @@ func cloneSource(task *service.RestTask, project *model.ProjectRef, config *mode
 	if variant == nil {
 		return errors.Errorf("couldn't find build variant '%v' in config", task.BuildVariant)
 	}
+	if mfest == nil && len(variant.Modules) != 0 { // manifest only required if variant has modules
```

Comment:
I think very old patches may not have a manifest, so if we don't have a manifest we should clone HEAD (it's wrong, but probably less wrong than erroring out)